Force quit on missing ACL

If email isn't found in survey ACLs:

```
android.view.InflateException: Binary XML file line #39 in com.google.android.ground:layout/home_screen_frag: Binary XML file line #39 in com.google.android.ground:layout/home_screen_frag: Error inflating class androidx.fragment.app.FragmentContainerView
                 Caused by: android.view.InflateException: Binary XML file line #39 in com.google.android.ground:layout/home_screen_frag: Error inflating class androidx.fragment.app.FragmentContainerView
                 Caused by: java.lang.IllegalStateException: ACL not found for email  in survey Demo survey
                 	at com.google.android.ground.model.Survey.getRole(Survey.kt:40)
                 	at com.google.android.ground.repository.UserRepository.canUserSubmitData(UserRepository.kt:70)
                 	at com.google.android.ground.ui.home.mapcontainer.HomeScreenMapContainerFragment.onCreate(HomeScreenMapContainerFragment.kt:77)
```

--- a/ground/src/main/java/com/google/android/ground/model/User.kt
+++ b/ground/src/main/java/com/google/android/ground/model/User.kt
@@ -17,7 +17,6 @@ package com.google.android.ground.model
 
 /** Represents a single application user. */
 data class User
-@JvmOverloads
 constructor(
   val id: String,
   val email: String,

--- a/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
+++ b/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
@@ -68,12 +68,11 @@ constructor(
    * Returns true if the currently logged in user has permissions to write data to the active
    * survey. If no survey is active at the moment, then it returns false.
    */
-  fun canUserSubmitData(): Boolean {
-    return try {
+  fun canUserSubmitData(): Boolean =
+    try {
       surveyRepository.activeSurvey?.getRole(currentUser.email) != Role.VIEWER
     } catch (e: IllegalStateException) {
       Timber.e(e, "Failed to check permissions for user $currentUser")
       false
     }
-  }
 }

--- a/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
+++ b/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
@@ -72,7 +72,7 @@ constructor(
     try {
       surveyRepository.activeSurvey?.getRole(currentUser.email) != Role.VIEWER
     } catch (e: IllegalStateException) {
-      Timber.e(e, "Failed to check permissions for user $currentUser")
+      Timber.e(e, "Error getting role for user $currentUser")
       false
     }
 }

--- a/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
+++ b/ground/src/main/java/com/google/android/ground/repository/UserRepository.kt
@@ -68,6 +68,12 @@ constructor(
    * Returns true if the currently logged in user has permissions to write data to the active
    * survey. If no survey is active at the moment, then it returns false.
    */
-  fun canUserSubmitData(): Boolean =
-    surveyRepository.activeSurvey?.getRole(currentUser.email) != Role.VIEWER
+  fun canUserSubmitData(): Boolean {
+    return try {
+      surveyRepository.activeSurvey?.getRole(currentUser.email) != Role.VIEWER
+    } catch (e: IllegalStateException) {
+      Timber.e(e, "Failed to check permissions for user $currentUser")
+      false
+    }
+  }
 }

--- a/ground/src/test/java/com/google/android/ground/repository/UserRepositoryTest.kt
+++ b/ground/src/test/java/com/google/android/ground/repository/UserRepositoryTest.kt
@@ -16,6 +16,7 @@
 package com.google.android.ground.repository
 
 import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.model.Role
 import com.google.android.ground.persistence.local.LocalValueStore
 import com.google.android.ground.persistence.local.stores.LocalUserStore
 import com.google.android.ground.system.NetworkManager
@@ -40,6 +41,7 @@ class UserRepositoryTest : BaseHiltTest() {
   @Inject lateinit var fakeAuthenticationManager: FakeAuthenticationManager
   @Inject lateinit var localUserStore: LocalUserStore
   @Inject lateinit var localValueStore: LocalValueStore
+  @Inject lateinit var surveyRepository: SurveyRepository
   @Inject lateinit var userRepository: UserRepository
   @Inject lateinit var fakeRemoteDataStore: FakeRemoteDataStore
 
@@ -90,5 +92,35 @@ class UserRepositoryTest : BaseHiltTest() {
     userRepository.clearUserPreferences()
 
     assertThat(localValueStore.lastActiveSurveyId).isEmpty()
+  }
+
+  @Test
+  fun `canUserSubmitData() when user has permissions returns true`() {
+    val user = FakeData.USER
+    val survey = FakeData.SURVEY.copy(acl = mapOf(Pair(user.email, Role.OWNER.toString())))
+    fakeAuthenticationManager.setUser(user)
+    surveyRepository.activeSurvey = survey
+
+    assertThat(userRepository.canUserSubmitData()).isTrue()
+  }
+
+  @Test
+  fun `canUserSubmitData() when user doesn't have permissions returns false`() {
+    val user = FakeData.USER
+    val survey = FakeData.SURVEY.copy(acl = mapOf())
+    fakeAuthenticationManager.setUser(user)
+    surveyRepository.activeSurvey = survey
+
+    assertThat(userRepository.canUserSubmitData()).isFalse()
+  }
+
+  @Test
+  fun `canUserSubmitData() when user email is empty returns false`() {
+    val user = FakeData.USER.copy(email = "")
+    val survey = FakeData.SURVEY.copy(acl = mapOf(Pair("user@gmail.com", Role.OWNER.toString())))
+    fakeAuthenticationManager.setUser(user)
+    surveyRepository.activeSurvey = survey
+
+    assertThat(userRepository.canUserSubmitData()).isFalse()
   }
 }

--- a/ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt
+++ b/ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceTaskFragmentTest.kt
@@ -153,7 +153,7 @@ class MultipleChoiceTaskFragmentTest :
     whenever(dataCollectionViewModel.isLastPosition(any())).thenReturn(true)
     setupTaskFragment<MultipleChoiceTaskFragment>(job, task)
 
-    hasButtons(ButtonAction.SKIP, ButtonAction.DONE)
+    assertFragmentHasButtons(ButtonAction.SKIP, ButtonAction.DONE)
   }
 
   @Test

